### ISSUE TITLE
Issue with has_aux in ArmijoSGD

### ISSUE THREAD

**Issue by adamian98** [google/jaxopt#329]
In the update function for ArmijoSGD, the value of aux is currently not updated:
```
next_state = ArmijoState(iter_num=state.iter_num+1,
                             error=error,
                             value=f_next,
                             aux=state.aux,
                             stepsize=stepsize,
                             velocity=next_velocity)
```

I think that in order to update the value of aux, the fourth line should be `aux=aux` instead of `aux=state.aux`.

**Comment by mblondel** [comment]
Good catch, agreed!

**Comment by Algue-Rythme** [comment]
Fix by #343
